pkg/applicationmanager/controllers/applicationbackup.go

```diff
@@ -979,6 +979,17 @@ func (a *ApplicationBackupController) backupResources(
 	backup *stork_api.ApplicationBackup,
 ) error {
 	var err error
+	var resourceTypes []metav1.APIResource
+	// Listing all resource types
+	if len(backup.Spec.ResourceTypes) != 0 {
+		optionalResourceTypes := []string{}
+		resourceTypes, err = a.resourceCollector.GetResourceTypes(optionalResourceTypes, true)
+		if err != nil {
+			log.ApplicationBackupLog(backup).Errorf("Error getting resource types: %v", err)
+			return err
+		}
+	}
+
 	// Always backup optional resources. When restorting they need to be
 	// explicitly added to the spec
 	objectMap := stork_api.CreateObjectsMap(backup.Spec.IncludeResources)
```

Comment:
I think, backup.Spec.ResourceType list and backup.Spec.IncludeResources list are mutual exclusive. We might need to add a check such that both are not set.
Also, While we should pass includeResource list to GetResourcesForType() as it is based on resource type.